scripts/docs_update_release.py

```diff
@@ -0,0 +1,224 @@
+                        </button>
+                    </form>
+                  </div>"""
+
+    new_entry_online = f"""            <div class="download-button">
+                    <form class="download-form" method="get" action="https://dl.espressif.com/dl/idf-installer/esp-idf-tools-setup-online-{online_installer_version}.exe">
+                        <button class="button-online">
+                            <i class="fa fa-download" aria-hidden="true"></i>
+                            <div>Universal Online Installer {online_installer_version}</div>
+                            <div>Windows 10, 11</div>
+                            <div>Size: {installer_size}</div>
+                        </button>
+                    </form>
+                </div>"""
+    
+    if installer_type == 'espressif-ide':
+        return new_entry_ide
+    elif installer_type == 'offline':
+        return new_entry_offline
+    elif installer_type == 'online':
+        return new_entry_online
+
+
+def update_index():
+    """Update the index.html file with the new release of the installer"""
+    try:
+        with open(path.abspath(INDEX_PATH), "r") as index_file:
+            index_lines = index_file.readlines()
+    except FileNotFoundError as e:
+        print(f"Error opening file {INDEX_PATH} - {e}")
+        raise SystemExit(1)
+
+    # find every element with the class "download-button"
+    elements = []
+    for i in range(0, len(index_lines)):
+        if index_lines[i].strip() == '<div class="download-button">':
+            elements.append([i, index_lines[i:i+10]])
+            i += 10 # skip the next 10 lines (the length of the element with the class "download-button")
+
+    # choose the elements that contain the installer type
+    selected_elements = []
+    for element in elements:
+        if any(f'{installer_type}' in element_line for element_line in element[1]):
+            selected_elements.append(element)
+
+    print(f"Found {len(selected_elements)} elements with the installer type {installer_type}")
+
+
+    def _replace_installer_button(element_to_replace:list) -> str:
+        """Replace the first occurrence of the installer button with the new one"""
+        element_to_replace = ''.join(element_to_replace[1])
+        print(f"This element will be replaced:\n{element_to_replace}")
+
+        index_data = ''.join(index_lines)
+        return index_data.replace(element_to_replace, _resolve_installer_type())
+
+
+    # replace the first occurrence of the offline installer button
+    if installer_type == 'offline':
+        element_to_replace = None
+        for selected_element in selected_elements:
+            for element_line in selected_element[1]:
+                if f'{idf_version[0]}.{idf_version[1]}' in element_line:
+                    element_to_replace = selected_element
+                    break
+        if element_to_replace:
+            new_index_data = _replace_installer_button(element_to_replace)
```

Comment:
You can drop `element_to_replace` variable here, like this: 
```suggestion
    if installer_type == 'offline':
        for selected_element in selected_elements:
            for element_line in selected_element[1]:
                if f'{idf_version[0]}.{idf_version[1]}' in element_line:
                    new_index_data = _replace_installer_button(element_to_replace)
                    break
```

Reply:
Thank you @peterdragun for your review!

The else statement of this condition (`if element_to_replace:`) is needed to decide if there is an element that can be replaced or if another one should be appended. Otherwise, both actions will be performed.

Reply:
That is right, but I will probably stick to the original solution.

The break is because of the structure of the element:
`<div class="download-button">
                    <form method="get" action="https://dl.espressif.com/dl/idf-installer/esp-idf-tools-setup-offline-5.2.1.exe">
                        <button class="button-offline">
                            <i class="fa fa-download" aria-hidden="true"></i>
                            <div>ESP-IDF v5.2.1 - Offline Installer</div>
                            <div>Windows 10, 11</div>
                            <div>Size: 1.5 GB</div>
                        </button>
                    </form>
                  </div> `

There are two places to find the IDF version and trigger the if statement.
- `<form method="get" action="https://dl.espressif.com/dl/idf-installer/esp-idf-tools-setup-offline-5.2.1.exe">`
- `<div>ESP-IDF v5.2.1 - Offline Installer</div>`

Reply:
Sure! If the solution works, I am okay with that. Let's not overcomplicate it.